File: smp2017-web-Darshit/CS50_Solutions/pset6/caesar.py
import cs50
import sys

if len(sys.argv) is not 2:
    print('Please provide the key')
    exit(1)
else:
    key = int(sys.argv[1])
    print("plaintext: ", end="")
    ptext = cs50.get_string();
    ctext = ""
    for char in ptext:
        if char.isalpha():
            if char.isupper():
                x = (ord(char) + key - 65) % 26 + 65
            elif char.islower():
                x = (ord(char) + key - ord('a')) % 26 + ord('a')
            c = chr(x)
            ctext += c
        else:
            ctext += char
    print("ciphertext:",ctext)      
    exit(0)

File: smp2017-web-Rohit/CS50/PSET-2/caesar.c
#include<stdio.h>
int main(void)
{
    int i,k;
    char q[20];
    printf("Key: ");
    scanf("%d",&k);
    printf("Plaintext: ");
    scanf("%s",q);
    char p[20];
    if(k>26)
    {
        k=k%26;
    }
    for(i=0;q[i]!='\0';i++)
    {
       p[i]=k+q[i];
       if(((p[i]>90)||(k>=26))&&(q[i]<91))
       {
           p[i]=p[i]-90+64;
       }
       else if(((p[i]>122)||(k>=26))&&(q[i]>96))
       {
           p[i]=p[i]-122+96;
       }
    }
    p[i]='\0';
    printf("Ciphertext: %s\n",p);
    return 0;
    
}
